p2p/host/autorelay/relay_finder.go

```diff
@@ -261,9 +266,12 @@ func (rf *relayFinder) runScheduledWork(ctx context.Context, now time.Time, sche
 	}
 
 	if now.After(scheduledWork.nextAllowedCallToPeerSource) {
-		scheduledWork.nextAllowedCallToPeerSource = scheduledWork.nextAllowedCallToPeerSource.Add(rf.conf.minInterval)
 		select {
 		case peerSourceRateLimiter <- struct{}{}:
+			scheduledWork.nextAllowedCallToPeerSource = now.Add(rf.conf.minInterval)
+			if scheduledWork.nextAllowedCallToPeerSource.Before(nextTime) {
+				nextTime = scheduledWork.nextAllowedCallToPeerSource
+			}
 		default:
 		}
 	}
```

Comment:
You will need to update scheduledWork.nextAllowedCallToPeerSource in the else clause to this if condition like this: 

```
if now.After(scheduledWork.nextAllowedCallToPeerSource) {
		scheduledWork.nextAllowedCallToPeerSource = scheduledWork.nextAllowedCallToPeerSource.Add(rf.conf.minInterval)
		select {
		case peerSourceRateLimiter <- struct{}{}:
			scheduledWork.nextAllowedCallToPeerSource = now.Add(rf.conf.minInterval)
			if scheduledWork.nextAllowedCallToPeerSource.Before(nextTime) {
				nextTime = scheduledWork.nextAllowedCallToPeerSource
			}
		default:
		}
}
else {
	if scheduledWork.nextAllowedCallToPeerSource.Before(nextTime) {
				nextTime = scheduledWork.nextAllowedCallToPeerSource
			}
}
```

Reply:
We need this because consider the following scenario:

minInterval = 70s
nextRefresh = 60s

Say the call to peerSource returns 1 candidate. Closes the channel and we want to call peer source again. 
This call won't be allowed immediately because minInterval is 70s. 
The nextTime set in scheduledWork will be 60s for nextRefresh. 

After 60s for nextRefresh scheduledWork this function will skip updating nextTime to scheduledWork.nextAllowedCallToPeerSource(10 seconds from now) and schedule the next run for 60s later. 

Am I missing something here?

Reply:
> Am I missing something here?

Nope! excellent point :)